reconcile/utils/gql.py

```diff
@@ -1,24 +1,27 @@
-import contextlib
-import json
 import logging
-import os
 import textwrap
-from typing import Set, Any
+from typing import Set, Any, Optional, Dict
 
 from urllib.parse import urlparse
 
 import requests
 
 from sretoolbox.utils import retry
-from graphqlclient import GraphQLClient
+
 from sentry_sdk import capture_exception
 
 from reconcile.utils.config import get_config
 from reconcile.status import RunningState
 
+from gql import gql, Client
+from gql.transport.requests import RequestsHTTPTransport
+from gql.transport.exceptions import TransportQueryError
+from gql.transport.requests import log as requests_logger
+
+from threading import Lock
 
-_gqlapi = None
 
+_gqlapi = None
```

Comment:
I agree that this could probably be done differently, but it also isn't strictly in the scope of this PR, so maybe we take this up in a separate ticket and PR?